src/core/fetchers/steering_manifest/steering_manifest_fetcher.ts

```diff
@@ -0,0 +1,184 @@
+export interface ISteeringManifestFetcherSettings {
+  /** Maximum number of time a request on error will be retried. */
+  maxRetryRegular : number | undefined;
+  /** Maximum number of time a request be retried when the user is offline. */
+  maxRetryOffline : number | undefined;
+}
+
+/**
+ * Class allowing to facilitate the task of loading and parsing a Content
+ * Steering Manifest, which is an optional document associated to a content,
+ * communicating the priority between several CDN.
+ * @class SteeringManifestFetcher
+ */
+export default class SteeringManifestFetcher {
+  private _settings : ISteeringManifestFetcherSettings;
+  private _pipelines : ITransportSteeringManifestPipeline;
+
+  /**
+   * Construct a new SteeringManifestFetcher.
+   * @param {Object} pipelines - Transport pipelines used to perform the
+   * Content Steering Manifest loading and parsing operations.
+   * @param {Object} settings - Configure the `SteeringManifestFetcher`.
+   */
+  constructor(
+    pipelines : ITransportSteeringManifestPipeline,
+    settings : ISteeringManifestFetcherSettings
+  ) {
+    this._pipelines = pipelines;
+    this._settings = settings;
+  }
+
+  /**
+   * (re-)Load the Content Steering Manifest.
+   * This method does not yet parse it, parsing will then be available through
+   * a callback available on the response.
+   *
+   * You can set an `url` on which that Content Steering Manifest will be
+   * requested.
+   * If not set, the regular Content Steering Manifest url - defined on the
+   * `SteeringManifestFetcher` instanciation - will be used instead.
+   *
+   * @param {string|undefined} url
+   * @param {Function} onRetry
+   * @param {Object} cancelSignal
+   * @returns {Promise}
+   */
+  public async fetch(
+    url : string,
+    onRetry : (error : IPlayerError) => void,
+    cancelSignal : CancellationSignal
+  ) : Promise<ISteeringManifestParser> {
+    const pipelines = this._pipelines;
+    const backoffSettings = this._getBackoffSetting((err) => {
+      onRetry(errorSelector(err));
+    });
+    const callLoader = () => pipelines.loadSteeringManifest(url, cancelSignal);
+    const response = await scheduleRequestPromise(callLoader,
+                                                  backoffSettings,
+                                                  cancelSignal);
+    return (onWarnings : ((error : IPlayerError[]) => void)) => {
+      return this._parseSteeringManifest(response, onWarnings);
+    };
+  }
+
+  /**
+   * Parse an already loaded Content Steering Manifest.
+   *
+   * This method should be reserved for Content Steering Manifests for which no
+   * request has been done.
+   * In other cases, it's preferable to go through the `fetch` method, so
+   * information on the request can be used by the parsing process.
+   * @param {*} steeringManifest
+   * @param {Function} onWarnings
+   * @returns {Observable}
+   */
+  public parse(
+    steeringManifest : unknown,
```

Comment:
we don't have a type yet for the steeringManifest ?

Reply:
We do, but `parse` take in input the result of the request, not the parsed one.

In most cases, it is a string or an object, but because following this type around could be hard (didn't check for that one right now, but it definitely was for segment requests as the types might be different based on the type of media and the same thing for the Manifest request where the type depend on the transport used - we could by the way consider the Content Steering Manifest the same way) and because we're never sure about the format of a requested object anyway, I preferred using `unknown` here. Type-checks are then performed at runtime in the parsing function - which may anyway be necessary after an external request